passkey-authenticator/src/authenticator/get_assertion.rs

```diff
@@ -69,13 +70,12 @@ where
         // 7. Collect user consent if required. This step MUST happen before the following steps due
         //    to privacy reasons (i.e., authenticator cannot disclose existence of a credential
         //    until the user interacted with the device):
-        let flags = self.check_user(&input.options).await?;
+        let flags = self
+            .check_user(&input.options, maybe_credential.as_ref().ok().cloned())
```

Comment:
**Thought:** Would it make sense to send in the slice of all possible credentials here? If `UserValidation` is done correctly it would give the app the possibility to select the passkey directly from this API. The default implementation could stick to the first one in the list.

Reply:
I actually did consider adding that as part of https://github.com/bitwarden/passkey-rs/pull/11 but I decided to wait because it would require changing the return value of `check_user`, which at the moment is used for both attestation and assertion. I wanted something a bit more statically typed, meaning either breaking up the `check_user` function or making the `UserValidationMethod` a bit more advanced by creating two implementations of it `UserValidationMethodWithDisplay` and one `UserValidationMethodWithoutDisplay`. 

We've temporarily solved this outside this library because I wanted to keep the changes to `passkey-rs` as small as possible.

Reply:
Yeah we've handled this outside the library as well. I guess I'm hoping to improve on the API to not need to think about how to handle these things outside the library. That unfortunately gets complicated in a multi-language setup when using rust 😅.

It could maybe be done by adding a `Context` parameter letting the the implementor know which context this is being called from. Like that they can decide to have it be with or without a display.